clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -4337,25 +4337,22 @@ public DescribeMetadataQuorumResult describeMetadataQuorum(DescribeMetadataQuoru
         final Call call = new Call(
                 "describeMetadataQuorum", calcDeadlineMs(now, options.timeoutMs()), provider) {
 
-            private QuorumInfo createQuorumResult(final DescribeQuorumResponse response) {
-                Integer partition = 0;
-                String topicName = response.getTopicNameByIndex(0);
-                Integer leaderId = response.getPartitionLeaderId(topicName, partition);
+            private QuorumInfo createQuorumResult(final DescribeQuorumResponseData.PartitionData partition) {
                 List<QuorumInfo.ReplicaState> voters = new ArrayList<>();
                 List<QuorumInfo.ReplicaState> observers = new ArrayList<>();
-                response.getVoterInfo(topicName, partition).forEach(v -> {
+                partition.currentVoters().forEach(v -> {
                     voters.add(new QuorumInfo.ReplicaState(v.replicaId(),
                                 v.logEndOffset(),
                                 OptionalLong.of(v.lastFetchTimestamp()),
                                 OptionalLong.of(v.lastCaughtUpTimestamp())));
```

Comment:
When `lastFetchTimestamp` or `lastCaughtUpTimestamp` are not provided (equals to -1), don't we want to return an empty option instead of returning an option containing -1?

Reply:
I went back and forth between that and ended up returning a -1 optional here. I now remember that the original intention was to have an empty optional. Will address this.